src/timepicker/timepicker.js

```diff
@@ -165,7 +165,7 @@ angular.module('ui.bootstrap.timepicker', [])
     var hours = +$scope.hours;
     var valid = $scope.showMeridian ? hours > 0 && hours < 13 :
       hours >= 0 && hours < 24;
-    if (!valid) {
+    if (!valid || $scope.hours === '') {
```

Comment:
Just a tip, the empty string is falsey so this and all your other checks of this nature can be simplified.